app/controllers/memberships_controller.rb

```diff
@@ -4,7 +4,7 @@ class MembershipsController < ApplicationController
 
   def create
     @group = Group.find(params[:group_id])
-    if @group.user_can_join?(current_user)
+    unless @group.user_can_join?(current_user)
```

Comment:
rubocop should scream about this :) 
in the sake of our and your sanity, never use the `unless .... else ... end` construct :) if you need to reverse the logic reorganize it into an `if ... else ... end` Or do an early return.